### PR TITLE
[Feature][MRV2] Support DSA decode ACLGraph

### PR DESCRIPTION
**Pull request by pgzddxx**
## What this PR does

This PR adds DSA decode ACLGraph support in ModelRunner V2 (FULL_DECODE_ONLY mode).

The generic build_for_cudagraph_capture() does not populate DSA-specific request-level metadata during graph capture, causing incomplete attention metadata. Changes:
1. Override build_for_cudagraph_capture() in AscendDSAMetadataBuilder, delegating to build() with capture-appropriate defaults.
2. Unify all backends to call build_for_cudagraph_capture() during capture; DSA uses its override via polymorphism, no special-casing.
3. Global rename num_reqs_actual → num_actual_reqs for consistency with num_actual_tokens, across all related files.

## Does this PR introduce user-facing change?

No. 

## How was this patch tested

- Unit tests pass (DSA attention tests).
- A3 TP4 runtime validation in FULL_DECODE_ONLY mode: graph capture and inference verified.

- vLLM version: v0.27.1
- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

File: vllm_ascend/worker/v2/model_runner.py [hunk 1]
```diff
@@ -580,6 +612,16 @@ def prepare_inputs(  # type: ignore[misc]
...
             )
+
+            if batch_desc.cg_mode == CUDAGraphMode.FULL and self._use_dsa_decode_aclgraph:
+                self._prepare_dsa_full_graph_padding_inputs(
+                    num_reqs=num_reqs,
+                    num_reqs_padded=num_reqs_padded,
+                    num_tokens=num_tokens,
+                    num_tokens_padded=num_tokens_after_padding,
+                    query_start_loc_np=query_start_loc_np,
+                )
```
Comment (yiz-liu) [inline vllm_ascend/worker/v2/model_runner.py]: Why is it that the other attn backends don't need this specific reset for padding tokens?

File: vllm_ascend/worker/v2/model_runner.py [hunk 2]
```diff
@@ -771,6 +814,52 @@ def _pad_query_start_loc_for_fia(
...
+        assert padding_start == num_tokens
+        assert padding_end == num_tokens_padded
+
+        padding_query_lens_np = np.diff(query_start_loc_np[num_reqs : num_reqs_padded + 1]).astype(np.int32, copy=False)
+        assert np.all(padding_query_lens_np > 0)
+        self.input_buffers.seq_lens_np[num_reqs:num_reqs_padded] = padding_query_lens_np
+        async_copy_to_gpu(
+            padding_query_lens_np,
+            out=self.input_buffers.seq_lens[num_reqs:num_reqs_padded],
+        )
```
Comment (yiz-liu) [inline vllm_ascend/worker/v2/model_runner.py]: Why do we put `query_lens` into `seq_lens`?

File: vllm_ascend/attention/dsa_v1.py [hunk 3]
```diff
@@ -395,6 +395,24 @@ def __init__(
...
+        self.decode_query_start_loc = torch.zeros(
+            scheduler_config.max_num_seqs + 1,
+            dtype=torch.int32,
+            device=self.device,
+        )
+        self.decode_seq_lens = torch.zeros(
+            scheduler_config.max_num_seqs,
+            dtype=torch.int32,
+            device=self.device,
+        )
```
Comment (yiz-liu) [inline vllm_ascend/attention/dsa_v1.py]: Please see if we can refactor these buffers to model runner, like every other buffer.

**Review by yiz-liu — COMMENTED** [review]
Basically, my concern is that we have implemented too much if-else for DSA, which is not acceptable.

**Comment by li1how** [comment]
This PR can be simplified to only:

1. Use the regular DSA metadata builder during graph capture and share metadata across cache groups.
2. Pass the actual request count separately from the padded request count.

The duplicate `positions` buffer and all DSA-specific padding/prefill logic in `model_runner.py` can be removed.

**Comment by li1how** [comment]
Check if there are any UT modifications.

File: vllm_ascend/attention/dsa_v1.py [hunk 4]
```diff
@@ -507,7 +524,7 @@ def build(
         **kwargs,
     ) -> AscendDSAMetadata:
         num_reqs = common_attn_metadata.num_reqs
-        num_reqs_actual = kwargs.get("num_reqs_actual")
+        num_actual_reqs = kwargs.get("num_actual_reqs")
```
Comment (weiguihua2) [inline vllm_ascend/attention/dsa_v1.py:527]: Also modify the num_reqs_actual of model runner v1.

**Comment by pgzddxx** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

**Review by yiz-liu — APPROVED** [review]

**Review by ningjingbengxiaohai — APPROVED** [review]
